spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/JdbcConfiguration.java

```diff
@@ -103,9 +103,31 @@ public JdbcCustomConversions jdbcCustomConversions() {
 	 */
 	@Bean
 	public JdbcAggregateOperations jdbcAggregateOperations(ApplicationEventPublisher publisher,
-			RelationalMappingContext context, JdbcConverter converter, NamedParameterJdbcOperations operations) {
-		DataAccessStrategy dataAccessStrategy = new DefaultDataAccessStrategy(new SqlGeneratorSource(context), context,
-				converter, operations);
+			RelationalMappingContext context, JdbcConverter converter, NamedParameterJdbcOperations operations,
+			DataAccessStrategy dataAccessStrategy) {
 		return new JdbcAggregateTemplate(publisher, context, converter, dataAccessStrategy);
 	}
+
+	/**
+	 * Register a {@link DataAccessStrategy} as a bean for reuse in the {@link JdbcAggregateOperations} and the
+	 * {@link RelationalConverter}.
+	 *
+	 * @param context
+	 * @param converter
+	 * @param operations
+	 * @return
+	 */
+	@Bean
+	public DataAccessStrategy getDataAccessStrategy(RelationalMappingContext context, JdbcConverter converter,
+			NamedParameterJdbcOperations operations) {
+
+		DefaultDataAccessStrategy defaultDataAccessStrategy = new DefaultDataAccessStrategy(new SqlGeneratorSource(context),
+				context, converter, operations);
+
+		if (converter instanceof BasicJdbcConverter) {
```

Comment:
We need to reconsider this one. We should not alter beans in a @Bean method. The need for a relation resolver on converter level makes it obvious that we create a cycle during runtime.